ai21/models/ai21_base_model.py

```diff
@@ -0,0 +1,45 @@
+import re
+from typing import Any, Dict
+
+from pydantic import BaseModel, ConfigDict
+
+from ai21.models.pydantic_compatibility import IS_PYDANTIC_V2
+
+if IS_PYDANTIC_V2:
+    from pydantic.alias_generators import to_camel
+
+
+def to_camel_case(snake: str) -> str:
+    if re.match("^[a-z]+[A-Za-z0-9]*$", snake) and not re.search(r"\d[a-z]", snake):
+        return snake
+
+    pascal = re.sub("([0-9A-Za-z])_(?=[0-9A-Z])", lambda m: m.group(1), snake.title())
+    return re.sub("(^_*[A-Z])", lambda m: m.group(1).lower(), pascal)
+
+
+class AI21BaseModel(BaseModel):
+    if IS_PYDANTIC_V2:
+        model_config = ConfigDict(
+            alias_generator=to_camel,
+            populate_by_name=True,
+            protected_namespaces=(),
+        )
+    else:
+
+        class Config:
+            alias_generator = to_camel_case
+            allow_population_by_field_name = True
```

Comment:
Why do we need `allow_population_by_field_name `?

Reply:
Without this configuration setting, you can only populate fields by the alias name, and not by the model attribute name.
For example, the following line:
```python
penalty = Penalty(scale=0.5, apply_to_whitespaces=True)
```
will not populate the field `apply_to_whitespaces`. Only this will work:
```python
penalty = Penalty(scale=0.5, applyToWhitespaces=True)
```

To allow populating by model attribute name, we need to set `allow_population_by_field_name` to True.

Reply:
Use hybrid `Field` as discussed

Reply:
Done, no need for a special hybrid field, we're using params that exist in both versions